src/main/java/org/gelecekbilimde/scienceplatform/post/controller/CategoryController.java

```diff
@@ -32,10 +29,29 @@ SuccessResponse<List<CategoryResponse>> getCategoryList() {
 		return SuccessResponse.success(categoryResponses);
 	}
 
+	@GetMapping("/{id}")
+	SuccessResponse<CategoryResponse> getCategoryById(@PathVariable Long id) {
+		Category category = categoryService.getCategory(id);
+		CategoryResponse categoryResponse = categoryToResponseMapper.map(category);
+		return SuccessResponse.success(categoryResponse);
+	}
+
 	@PostMapping("/create")
 	SuccessResponse<Void> createCategory(@RequestBody @Valid CategoryCreateRequest request) {
 		categoryService.createCategory(request);
 		return SuccessResponse.success();
 	}
 
+	@PutMapping("/{id}/update")
+	SuccessResponse<Void> updateCategory(@PathVariable Long id, @RequestBody @Valid CategoryUpdateRequest request) {
+		categoryService.updateCategory(id, request);
+		return SuccessResponse.success();
+	}
+
+	@DeleteMapping("/{id}/delete")
```

Comment:
@aleyna-yildizli `DELETE` HTTP Metodu aslında delete ettiğimizi anlatabiliyor bu yüzden davranışı endpointlerde yazmamaya çalışalım. Ek olarak `@RequestMapping`'de tanımlamak yerine aşağıdaki gibi tanımlayabiliriz. Bu sayede listeleme endpotine `/categories` yazabiliriz.
```suggestion
	@DeleteMapping("/category/{id}")
```